client/src/pages/inventory/VariantCreate.tsx

```diff
@@ -1,51 +1,209 @@
-const VariantCreateForm = ({ handleClose }: VariantCreateDialogProps) => (
-  <>
-    <DialogTitle>Create a New Variant</DialogTitle>
-    <DialogContent>
-      <Spacer />
-      <Grid container spacing={2}>
-        <Grid item xs={12}>
-          <TextField
-            fullWidth
-            label="Variant Descriptor (i.e 500ml, blue)"
-            variant="outlined"
-          />
-        </Grid>
-        <Grid item xs={6}>
-          <TextField fullWidth label="Price" variant="outlined" />
-        </Grid>
-        <Grid item xs={6}>
-          <TextField fullWidth label="Barcode" variant="outlined" />
-        </Grid>
-      </Grid>
-    </DialogContent>
-    <DialogActions>
-      <Button color="inherit">Cancel</Button>
-      <Button>Create</Button>
-    </DialogActions>
-  </>
-);
-
-export default function VaraintCreateDialog({
+const defaultProps = {
+  variant: undefined,
+  isCreateVariantLoading: false,
+  hasCreateVariantError: false,
+  isDeleteVariantLoading: false,
+  hasDeleteVariantError: false,
+};
+
+const VariantCreateDialog: React.FC<VariantCreateDialogProps> = ({
   handleClose,
-}: VariantCreateDialogProps) {
-  return <VariantCreateForm handleClose={handleClose} />;
-}
+  isOpen,
+  handleCreate,
+  handleDelete,
+  variant,
+  isCreateVariantLoading,
+  hasCreateVariantError,
+  isDeleteVariantLoading,
+  hasDeleteVariantError,
+}) => {
+  const { control, handleSubmit, reset } = useForm<PostVariantBody>();
+
+  useEffect(() => {
+    reset({
+      description: variant?.description,
+      price: penniesToFloat(variant?.price || 0),
+      barcode: variant?.barcode,
+    });
+  }, [variant]);
+
+  const closeActions = () => {
+    handleClose();
+    reset();
+  };
+
+  return (
+    <Dialog open={isOpen} onClose={closeActions}>
+      <DialogTitle>
+        {variant ? 'Update a variant' : 'Create a new variant'}
+      </DialogTitle>
+      <DialogContent>
+        {hasCreateVariantError && (
+          <Alert severity="error">
+            <AlertTitle>An unexpected error has occured</AlertTitle>
+            Something went wrong while creating a variant
+          </Alert>
+        )}
+        {hasDeleteVariantError && (
+          <Alert severity="error">
+            <AlertTitle>An unexpected error has occured</AlertTitle>
+            Something went wrong while deleting a variant
+          </Alert>
+        )}
+        <Spacer />
+        <Grid container spacing={2}>
+          <Grid item xs={12}>
+            <Controller
+              name="description"
+              control={control}
+              rules={{
+                required: true,
+                minLength: 1,
+              }}
+              render={({
+                field: { onChange, value },
+                fieldState: { invalid, error },
+              }) => (
+                <TextField
+                  fullWidth
+                  label="Variant Descriptor (i.e 500ml, blue)"
```

Comment:
So _i.e._ in English means "in another word", what you are looking for is _e.g._. But I recommend that you just omit it, people will understand.

https://www.warnell.uga.edu/sites/default/files/inline-files/ieVeg.pdf

TL;DR:

> I find that sometimes students are unsure about when to use the abbreviations “i.e.” and 
“e.g.” and the difference between the two. They are not interchangeable; each has its own 
meaning and usage. The abbreviation “i.e.” stands for id est, which is Latin for “that is.” The 
abbreviation “e.g.” stands for the Latin phrase exempli gratia, meaning “for example.”